figenv.py

```diff
@@ -81,14 +87,21 @@ def __getattr__(cls, name):
         if (not load_all and name not in cls._dict) or (name not in cls._dict and prefix + name not in os.environ):
             raise AttributeError(f"type object {cls.name} has no attribute '{name}'")
 
-        if prefix + name in os.environ:
+        value = cls._dict.get(name, _MISSING)
+
+        override_via_environment = True
+        if callable(value) and getattr(value, "_strict", False):
+                override_via_environment = False
+
+        if override_via_environment and prefix + name in os.environ:
             value = os.environ[prefix + name]
-        else:
-            value = cls._dict[name]
 
         if callable(value):
             value = value(cls)
 
+        if value is _MISSING:
+            raise AttributeError(f"type object {cls.name} has no attribute '{name}'")
```

Comment:
In what instance does this actually get hit?

Reply:
It should get hit if the name is not in the class dict nor in the environment